templates/workloads/partials/cf/service-base-properties.yml

```diff
@@ -20,17 +20,26 @@ PropagateTags: SERVICE
 LaunchType: FARGATE
 NetworkConfiguration:
   AwsvpcConfiguration:
-    AssignPublicIp: ENABLED
+    {{- $assignPublicIP := "DISABLED" }}
+    {{- $subnets := "PrivateSubnets" }}
+    {{- if or (eq .Network.Placement "") (eq .Network.Placement "public") }}
+      {{- $assignPublicIP = "ENABLED" }}
+      {{- $subnets = "PublicSubnets" }}
+    {{- end }}
+    AssignPublicIp: {{$assignPublicIP}}
```

Comment:
Does this make sense to wrap in a template function for easier testing?

Reply:
Good idea!
I'll add an `AssignPublicIP` method to `NetworkOpts` to make this mess go away :)